src/regdesk/printer_manager.js

```diff
@@ -0,0 +1,66 @@
+import { LabelEpl, LP2844 } from 'WebZLP/src/LP2844'
+
+export class PrinterManager {
+    #devices = [];
+    #printers = [];
+
+    primaryPrinter = 0;
+    secondaryPrinter = 1;
+
+    #printerType = LP2844;
+
+    constructor() {
+    }
+
+    get printers() { return this.#printers; }
+
+    async printerConnect(e) {
+        if (!this.#devices.includes(e.device)) {
+            this.#devices.push(e.device);
+            let printer = new this.#printerType(e.device);
+            await printer.connect();
+            this.#printers.push(printer);
+            this.primaryPrinter = 0;
+        }
+    }
+
+    async printerDisconnect(e) {
+        const idx = this.#devices.findIndex(i => e.device == i);
+        if (idx > -1) {
+            const p = this.#printers[idx];
+            this.#devices.splice(idx, 1);
+            this.#printers.splice(idx, 1);
+            this.resetPrinterButtons();
+            await p.dispose();
+        }
+    }
+
+    async printerPair() {
+        try {
+            let device = await navigator.usb.requestDevice({
+                filters: [
+                    { vendorId: this.#printerType.usbVendorId }
+                ]
+            });
+
+            await this.printerConnect( { device : device} )
+        } catch (e) {
+            console.log("Failed to connect to printer!");
+            console.log(e);
+            return;
+        }
+    }
+
+    async printerReload(e) {
```

Comment:
I don't think you do anything with e? And it looks like this reloads all printers (plural)?

EDIT: Looks like this method isn't called, which is good because it didn't sit well with me haha, can it be deleted?

Reply:
It lives on in a slightly different form. We need a method to, on page load, reconnect to all of the printers and load them up so the user doesn't have to click all over the place if they accidentally refresh the page or something.

 I've got a TODO to make it a bit more clever-er and thus more reliable at coming back in a stable state, right now it's a race condition on what printer lands in what slot. We should fix that before we ship.